docs/native-modules-android.md

```diff
@@ -3,113 +3,124 @@ id: native-modules-android
+### Create A Custom Native Module File
 
-Create a new Java Class named `ToastModule.java` inside `android/app/src/main/java/com/your-app-name/` folder with the content below:
+The first step is to create a main custom native module file. Create a new Java class named CalendarModule.java inside `android/app/src/main/java/com/your-app-name/ folder`.
 
-```java
-// ToastModule.java
-
-package com.your-app-name;
+<figure>
+  <img src="/docs/assets/native-modules-android-add-class.png" width="700" alt="Image of adding a class called CalendarModule.java within the Android Studio." />
+  <figcaption>Image of how to add the CalendarModuleClass</figcaption>
+</figure>
 
-import android.widget.Toast;
+Then add the following content:
 
+```jsx
+package com.your-app-name; // replace com.your-app-name with your app’s name
 import com.facebook.react.bridge.NativeModule;
 import com.facebook.react.bridge.ReactApplicationContext;
 import com.facebook.react.bridge.ReactContext;
 import com.facebook.react.bridge.ReactContextBaseJavaModule;
 import com.facebook.react.bridge.ReactMethod;
-
 import java.util.Map;
 import java.util.HashMap;
 
-public class ToastModule extends ReactContextBaseJavaModule {
-  private static ReactApplicationContext reactContext;
+public class CalendarModule extends ReactContextBaseJavaModule {
+   CalendarModule(ReactApplicationContext context) {
+       super(context);
+   }
+}
+```
+
+As you can see, your `CalendarModule` class extends the `ReactContextBaseJavaModule`. For Android, most native modules consist of a class that extends `ReactContextBaseJavaModule` and implements the functionality required by Javascript.
 
-  private static final String DURATION_SHORT_KEY = "SHORT";
-  private static final String DURATION_LONG_KEY = "LONG";
+> Note: It is worth noting that technically Java classes can extend the `BaseJavaModule` class or implement the `NativeModule` interface to be considered a Native Module by React Native.
 
-  ToastModule(ReactApplicationContext context) {
-    super(context);
-    reactContext = context;
-  }
+> However we recommend that you use `ReactContextBaseJavaModule`, as shown above. `ReactContextBaseJavaModule` gives access to the `ReactApplicationContext` (RAC), which is useful for Native Modules that need to hook into activity lifecycle methods. Using `ReactContextBaseJavaModule` will also make it easier to make your native module type-safe in the future. For native module type-safety, which is coming in future releases, React Native looks at each native module's Javascript spec and generates an abstract base class that extends `ReactContextBaseJavaModule`.
+
+### Module Name
+
+All native modules in Android need to implement the `getName()` method. This method returns the string name of the native module which represents the class in JavaScript. The native module can then be accessed in JS via its name. For example, in the below code snippet, `getName` returns `"CalendarModule"`.
```

Comment:
What's the convention on writing method names: Do we always suffix with `()`?

Reply:
I am not sure, @rachelnabors  is there an opinion here? I removed () from all method names in the docs to be consistent for now!

Reply:
We use `()` to help distinguish methods and functions from objects at a glance. (Unless you can think of a reason not to!)

Reply:
just updated to include them